### DISCUSSION TITLE
Reading phpstan configuration

### DISCUSSION THREAD

**Discussion by nikophil** [rectorphp/rector#8379] [Q&A] [answered]
Hello,

is there a way to access phpstan's configuration from rector rule.

I'd like to leverage `phpstan-doctrine`, and its config `doctrine.objectManagerLoader`. I need to instantiate `PHPStan\Type\Doctrine\ObjectMetadataResolver` but I don't find any way to inject the loader into it:

```
    $rectorConfig->when(ObjectMetadataResolver::class)
        ->needs('$objectManagerLoader')
        ->giveConfig('%doctrine.objectManagerLoader%'); // I don't know what to put here
```

thanks for your answer

**Comment by samsonasik** [comment, marked as answer]
If it is part of phpstan service, you may need to pull from `\PHPStan\DependencyInjection\Container`, see this as reference 

https://github.com/rectorphp/rector-src/blob/f4257afb3c05217b13513c4708228a6a976bab88/packages/NodeTypeResolver/DependencyInjection/PHPStanServicesFactory.php#L26

**Reply by nikophil** [reply]
thanks!
